test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1731,6 +1731,20 @@ public static boolean indexExists(String index, Client client) {
         return getIndexResponse.getIndices().length > 0;
     }
 
+    public static void ensureIndexExists(String index) {
+        ensureIndexExists(index, SAFE_AWAIT_TIMEOUT.seconds(), SAFE_AWAIT_TIMEOUT.timeUnit());
+    }
+
+    public static void ensureIndexExists(String index, long timeout, TimeUnit unit) {
```

Comment:
I'd rather we just always used `SAFE_AWAIT_TIMEOUT` (I suspect the one case where we wait 30s below is a bug and could be 10s). But if we do need to expose the timeout to callers could we use `TimeValue` in the API rather than `long/TimeUnit`? I'll fix up the `safeGet` overload to use `TimeValue` too.

Reply:
I deleted the 30 seconds use cases. But still kept the method variant that takes a timeout parameter since the usage for remote cluster has a 120 seconds wait time. I suspect 120s is excessive. But probably justify to be longer than 10s.